### PR TITLE
Use async execution for xcodebuild commands

### PR DESCRIPTION
**Pull request by priitlatt**
## Description

Sometimes Flutter never receives the output of `xcodebuild -list` and `xcodebuild -showBuildSettings` commands while they have actually been completed underneath (no alive processes for `xcodebuild` are present). For unknown reasons, the issue does not reproduce when the commands are invoked with `runCheckedAsync` instead of `runCheckedSync`.

## Related Issues

#28415

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/test/project_test.dart [hunk 1]
```diff
@@ -264,10 +264,10 @@ void main() {
       });
       testWithMocks('from pbxproj and plist, by substitution', () async {
         final FlutterProject project = await someProject();
-        when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenReturn(<String, String>{
+        when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenAnswer((_) => Future<Map<String, String>>.value(<String, String>{
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/project_test.dart]:
```suggestion
        when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenAnswer((Invocation invocation) => Future<Map<String, String>>.value(<String, String>{
```

File: packages/flutter_tools/test/ios/cocoapods_test.dart [hunk 2]
```diff
@@ -152,12 +152,12 @@ void main() {
 
     testUsingContext('creates swift Podfile if swift', () async {
       when(mockXcodeProjectInterpreter.isInstalled).thenReturn(true);
-      when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenReturn(<String, String>{
+      when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenAnswer((_) => Future<Map<String, String>>.value(<String, String>{
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/ios/cocoapods_test.dart]:
```suggestion
      when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenAnswer((Invocation invocation) => Future<Map<String, String>>.value(<String, String>{
```

**Comment by jonahwilliams** [comment]
I'm ready to LGTM once the CLA is signed and the test is fixed

**Comment by jonahwilliams** [comment]
@priitlatt are you able to continue working on it or should we continue ourselves?

**Comment by priitlatt** [comment]
@jonahwilliams We were observing how this fix behaves in the production over the weekend and yesterday. At first it was very promising, but now it seems that it still occasionally gets stuck on `/usr/bin/xcodebuild -project /paht/to/Runner.xcodeproj -target Runner -showBuildSettings`.

Also, making `XcodeProjectInterpreter#getBuildSettings` async has proven to be a bit more involved than I initially thought it would be (this also causes the tests and some functionality to break). And on top of the complications, it doesn't even give the desired result.

But async version of `XcodeProjectInterpreter#getInfo` seems to solve at least half of the problem for now. So I'll revert changes related to `getBuildSettings` and perhaps we could move on with `getInfo` being async for now?

**Comment by jonahwilliams** [comment]
That sounds reasonable to me

**Comment by jonahwilliams** [comment]
cc @priitlatt what are your plans with this CL? If this improves the situation I would prefer to land it and continue addressing other forms of flakiness separately

**Comment by priitlatt** [comment]
@jonahwilliams Last I checked the Flutter build was failing and hence the delays, sorry about that. I'm good to go on with this and continue with `showBuildSettings` getting stuck separately. Also, I've signed the CLA in the meantime.

**Review by jonahwilliams — APPROVED** [review]
LGTM,

Thank you for the contribution!

**Comment by priitlatt** [comment]
I signed it!

**Comment by priitlatt** [comment]
I signed it!

**Comment by jonsamwell** [comment]
Is this in the stable branch yet?

**Pull request merged**